core/dbt/compilation.py

```diff
@@ -371,6 +371,18 @@ def _compile_node(
                 node,
             )
 
+        # relation_name is set at parse time, except for tests without store_failures,
+        # but cli param can turn on store_failures, so we set here.
+        if (
```

Comment:
The code looks fine, but should we have an associated unit test that captures this scenario?

Reply:
Yes, I'm working on a test. Should probably have set this to draft since I wasn't finished.

Reply:
Oh apologies for jumping the gun!